tempesta_fw/cfg.c

```diff
@@ -1517,61 +1530,69 @@ tfw_cfg_start_mods(const char *cfg_text, struct list_head *mod_list)
 
 err_recover_stop:
 	TFW_DBG2("stopping already stared modules\n");
-	MOD_FOR_EACH_REVERSE_FROM_CURR(tmp_mod, mod, mod_list) {
+	MOD_FOR_EACH_REVERSE_FROM_CURR(tmp_mod, mod, &tfw_cfg_mods) {
 		mod_stop(tmp_mod);
 	}
 
 err_recover_cleanup:
-	MOD_FOR_EACH_REVERSE(mod, mod_list) {
+	MOD_FOR_EACH_REVERSE(mod, &tfw_cfg_mods) {
 		spec_cleanup(mod->specs);
 	}
 
 	return ret;
 }
 
+int
+tfw_cfg_start(void)
```

Comment:
Original Cfg code mixes start/stop/restart activity with configuration. I did the first step in separating start/stop logic from configuration by moving sysctl routines out of cfg.c. I believe further steps must be taken to clean configuration code. The modules list must be handled outside cfg.c. So we control start/stop consistency **outside** cfg.c. I though it's obvious that calling start functions twice is wrong. Who in earth might wish to do so?